src/main/java/io/vertx/proton/impl/ProtonTransport.java

```diff
@@ -284,4 +284,9 @@ public void handle(Long event) {
       }
     }
   }
+
+  ProtonTransport setIdleTimeout(int idleTimeout) {
+    this.transport.setIdleTimeout(idleTimeout);
```

Comment:
No, I meant that I thought it wouldn't always work on the server side as intended given where the checking task is scheduled, but considering it again I believe I was wrong as I overlooked an aspect of how vertx/netty works.